frame/benchmarking/src/lib.rs

```diff
@@ -231,6 +225,98 @@ pub use v1::*;
 /// 	);
 /// }
 /// ```
+///
+/// ## Benchmark Function Generation
+///
+/// The benchmark function definition that you provide is used to automatically create a number
+/// of impls and structs required by the benchmarking engine. Additionally, a benchmark
+/// function is also generated that resembles the function definition you provide, with a few
+/// modifications:
+/// 1. The function name is transformed from i.e. `original_name` to `_original_name` so as not
+///    to collide with the struct `original_name` that is created for some of the benchmarking
+///    engine impls.
+/// 2. Appropriate `T: Config` and `I` (if this is an instance benchmark) generics are added to
+///    the function automatically during expansion, so you should not add these manually on
+///    your function definition (but you may make use of `T` and `I` anywhere within your
+///    benchmark function, in any of the three sections (setup, call, verification).
```

Comment:
This here is quite magical, the fact that you can just use `T` or `I` in the benchmarking code without declaring them. What if a user wants to attach a `where` clause to constrain these 2 type parameters? Do you simply put it in the `benchmark` attribute?

Reply:
You can specify it at the benchmarking module level https://paritytech.github.io/substrate/master/frame_benchmarking/v2/index.html#where-clause

Reply:
Yes it is a bit magical but they were also injected old syntax. I dont see an easy way to define them if we want to use a module and not a struct.  

The `where` clause is only supported for all benchmarks at once. So far this was not an issue since you normally want to benchmark all calls at once to generate a correct `weights.rs`.